Sources/Datadog/Kronos/KronosInternetAddress.swift

```diff
@@ -43,6 +43,37 @@ internal enum KronosInternetAddress: Hashable {
         }
     }
 
+    /// If the address is reserved for private internets (local / private IP).
+    var isPrivate: Bool {
+        guard let host = host else {
+            return false
+        }
+
+        switch self {
+        case .ipv6:
+            // Ref.:  https://datatracker.ietf.org/doc/html/rfc4193#section-3
+            //
+            // +--------+-+------------+-----------+----------------------------+
+            // | 7 bits |1|  40 bits   |  16 bits  |          64 bits           |
+            // +--------+-+------------+-----------+----------------------------+
+            // | Prefix |L| Global ID  | Subnet ID |        Interface ID        |
+            // +--------+-+------------+-----------+----------------------------+
+            //
+            // Local IP is expected to have FC00::/7 prefix (7 bits) and L byte set to 1,
+            // which effectively means `fd` prefix for local IPs.
+            return host.starts(with: "fd")
+        case .ipv4:
+            // Ref.: https://datatracker.ietf.org/doc/html/rfc1918#section-3
+            //
+            // Local IPs have predefined ranges:
+            // - class A: 10.0.0.0 — 10.255.255.255
+            // - class B: 172.16.0.0 — 172.31.255.255
+            // - class C: 192.168.0.0 — 192.168.255.255
+            let regex = #"^((10\.)|(172\.1[6-9]\.)|(172\.2[0-9]\.)|(172\.3[0-1]\.)|(192\.168\.))"#
+            return host.range(of: regex, options: .regularExpression) != nil
```

Comment:
As an alternative, we could use bit masks instead of regex. That would require reading IP bytes from [`sockaddr_in`](https://developer.apple.com/documentation/kernel/sockaddr_in). It could be easy for IPv4, but seems trickier for IPv6 ([`sockaddr_in6`](https://developer.apple.com/documentation/kernel/sockaddr_in6)). To avoid implementing raw memory binds, I just used existing `host` value and regex.

Effectively, the SDK resolves 3-5 IPs on startup, so performance is not a problem.

Reply:
i think we should cover [all the cases that Eskimo mentions here](https://developer.apple.com/forums/thread/663848) 
(`224.0.0.0/4` and `255.255.255.255` are currently uncovered)